packages/plugins/src/foundry/utils/trim.ts

```diff
@@ -0,0 +1,158 @@
+import { BuildInfo } from '@sphinx-labs/core'
+
+/**
+ * This function is used to trim down the build info file to exactly the fields we specify in our type.
+ * To do this, we use a template object which contains all the fields we want. We cannot use the typescript
+ * type itself since that doesn't exist at runtime. So, if we want to update the BuildInfo type, we also
+ * need to update the template below.
+ *
+ * This is kind of a foot gun since we could forget to update the template, but it's not that big of a
+ * concern since we almost never change the BuildInfo type. It's also non-trivial to resolve. We'd need to do
+ * something like autogenerate the BuildInfo type from the template defined here to really resolve it.
+ *
+ * @dev Note that this function does not implement any logic for trimming individual elements inside of arrays.
+ * It is only possible to select the entire array.
+ */
+export const trimObjectToType = <T>(obj: any, typeTemplate: T): T => {
+  // If the template is either an array then we just return since we don't do any filtering on arrays
+  if (Array.isArray(typeTemplate)) {
+    return obj
+  } else if (typeof typeTemplate !== 'object') {
+    // If the template is a primative type, then we just return so the value is included in the result
+    return obj
+  } else if (!typeTemplate) {
+    throw new Error('Cannot specify a typeTemplate that is null or undefined')
+  }
```

Comment:
This'll throw an error for null-ish `typeTemplate` values, like `0` and `''`

Reply:
That's actually not true because primative types that are falsy will be caught and returned in the if statement above, but we can specifically check for `null` here it is more specific.